### PR TITLE
Amorphous 'bars' interaction requires all checked parts unencumbered

### PR DESCRIPTION
**Pull request by NoddingCrow**
#### Summary

SUMMARY: Bugfixes "Amorphous Bars interaction correctly checks all body parts are unencumbered"

#### Purpose of change

Currently the check for amorphous characters to pass through metal bars is 
IF head AND torso AND (either foot) THEN FAIL - i.e. a character in full power armor can simply take off their helmet to pass the check, then somehow phase a full suit of tank plating through a three-inch gap. 
Players can no longer remove their boots to pass through bars in full plate armor.

Reproducible: Aquire character with AQUEOUS and AMORPHOUS traits, spawn metal_bars (t_bars) terrain, examine it. Don full power armor, examine it again. Remove _just_ the helmet, and examine it a third time.

#### Describe the solution
Iterative loop through all body parts when called, early-returning with message if _any_ are 10 encumb or higher.

#### Describe alternatives you've considered

Perhaps change this to a sum-total encumberance check - it wouldn't be outrageous to be able to force medium-encumberance gear through the bars with a time penalty. However, this works for now.

#### Testing

Code compiled, t_bars and t_window_bars (per other PR) examined by aqueous/amorphous character with and without encumbering gear, and again with amorphous trait removed to check the special action is cleared correctly. Checked with single pieces of encumbering gear in each checked slot, and combinations thereof.

### CODE REVIEW

**Comment by Fris0uman** [comment]
Wha about hands abd arms? You could unharcode this check by iterating over every bodyparts using `get_all_bodyparts()`.
Unhardcoding stuff is good and it would make a bit more sense if all parts were concerned byt the check.

**Comment by NoddingCrow** [comment]
Hi there, @Fris0uman 
I'm a _very_ junior C++ coder so I absolutely went for the simplest solution I could find, partially to fix this problem while working on my other PR and partially just proof-of-viability for running compilation.
That said, I agree with your assessment and if there's an easy way to do this I'll 100% investigate. Thanks for the tip!

**Comment by NoddingCrow** [comment]
oh dang, you right. That's super easy. One hell of a tip, there.

File: src/iexamine.cpp [hunk 1]
```diff
@@ -1295,15 +1295,13 @@ void iexamine::bars( player &p, const tripoint &examp )
...
-          10 ) ) { // Most likely places for rigid gear that would catch on the bars.
-        add_msg( m_info,
-                 _( "Your amorphous body could slip though the %s, but your cumbersome gear can't." ),
-                 here.tername( examp ) );
-        return;
+    for (const bodypart_id &bp : p.get_all_body_parts()) {
+        if (p.encumb(bp) >= 10) {
+            add_msg(m_info,
+                    _("Your amorphous body could slip though the %s, but your cumbersome gear can't."),
+                    here.tername(examp));
```
Comment (BrettDong) [inline src/iexamine.cpp:1302]:
```suggestion
    for( const bodypart_id &bp : p.get_all_body_parts() ) {
        if( p.encumb( bp ) >= 10 ) {
            add_msg( m_info,
                     _( "Your amorphous body could slip though the %s, but your cumbersome gear can't." ),
                     here.tername( examp ) );
```

**Review by BrettDong — CHANGES_REQUESTED** [review]
Have to follow C++ style convention in this repository to pass CI checks. You can run `make astyle` to format C++ source code before making a commit.

**Review by BrettDong — APPROVED** [review]

**Pull request merged**